File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    /* Input: prices = [10,1,5,6,7,1]
    Output: 6 
    Exp: buy for 1 (day 2) sell for 7 (day 5); 
    */
    public int maxProfit(int[] prices) {
        int ans = 0;
        int n = prices.length;
        int min = Integer.MAX_VALUE;
        for(int i=0; i<n; i++){
            min = Math.min(min, prices[i]);
            ans = Math.max(prices[i]-min, ans);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.kt
class Solution {
    var ans:Int = Integer.MAX_VALUE
    fun findMin(nums: IntArray): Int {

        minHelper(nums, 0, nums.size-1)
        return ans
    }

    fun minHelper(nums: IntArray, l: Int, r: Int){
        if(l>r){
            return
        }

        val mid = l+(r-l)/2
        if(nums[l]<=nums[mid] && nums[mid]<= nums[r]){
            ans = min(ans, nums[l])
        }else if(nums[l]<=nums[mid]){
            ans = min(ans, nums[l])
            minHelper(nums, mid+1, r)
        }else {
            ans = min(ans, nums[mid])
            minHelper(nums, l, mid-1)
        }

        return

    }
}
